### PR TITLE
fix(TDI-48569): salesforce dynamic schema

### PR DESCRIPTION
**Pull request by ozhelezniak-talend**
* support parent-child relation for dynamic schema
* like: SELECT Account.Name FROM Contact

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-48569

**What is the new behavior?**
https://jira.talendforge.org/browse/TDI-48569

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
Related PR:
https://github.com/Talend/components/pull/2132
https://github.com/Talend/components/pull/2133

### CODE REVIEW

**Review by undx — APPROVED** [review]
LGTM

**Review by undx — APPROVED** [review]
some comments but LGTM

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceInputReader.java [hunk 1]
```diff
@@ -56,36 +58,44 @@ protected Schema getSchema() throws IOException {
...
+                                    .findAny();
+                            schemaField = optionalField.orElse(null);
+                            // fix fullName if it doesn't match the real column name
+                            // case not sensitive
+                            fullName = optionalField
+                                    // we already checked that those parts are equal when ignore case
+                                    // possible problem when we replace not only simple name
+                                    .map(it -> fieldDescription.getFullName().replace(simpleName, it.name()))
+                                    // else value doesn't matter, we expect to handle next iteration instead
+                                    .orElse(null);
```
[the remark this answers is not in the document]
Comment (ozhelezniak-talend) [reply components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceInputReader.java:82]: I didn't want. The result of this block is the initialization of two variables. 
fullName and schemaField.
I can make the optional field and schema field into one statement, but as result, I either need to create a new Optional to wrap schemaField or add a new if-else block, and it'll make the code looks more complicated. I thought that creating a new optional that I already had is a bit wierd.

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceInputReader.java [hunk 2]
```diff
@@ -56,36 +58,44 @@ protected Schema getSchema() throws IOException {
...
-                                Object propValue = fieldProps.get(propName);
-                                if (propValue != null) {
-                                    field.addProp(propName, propValue);
-                                }
+                        Schema.Field field = new Schema.Field(fullName, schemaField.schema(), schemaField.doc(), schemaField.defaultVal());
+                        Map<String, Object> props = schemaField.getObjectProps();
+                        for (String propName : props.keySet()) {
+                            Object propValue = props.get(propName);
+                            if (propValue != null) {
+                                field.addProp(propName, propValue);
```
[the remark this answers is not in the document]
Comment (ozhelezniak-talend) [reply components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceInputReader.java:95]: I wanted to remain it look almost the same as it is in the guessing part.
Make the change as small as possible without refactoring. 
Maybe I can convert it into a stream, but it's the code that was here before and it allows us to do some references between it and the guessing part.

**Pull request merged**